lib/charms/mongodb/v1/mongos.py

```diff
@@ -171,30 +175,44 @@ def remove_shard(self, shard_name: str) -> None:
 
         Raises:
             ConfigurationError, OperationFailure, NotReadyError,
-            RemovePrimaryShardError
+            NotEnoughSpaceError
         """
-        sc_status = self.client.admin.command("listShards")
-
         # It is necessary to call removeShard multiple times on a shard to guarantee removal.
         # Allow re-removal of shards that are currently draining.
+        sc_status = self.client.admin.command("listShards")
```

Comment:
Can we also add a mandatory pre-flight check: Ensure the Balancer Process is Enabled?
https://www.mongodb.com/docs/manual/tutorial/remove-shards-from-cluster/#ensure-the-balancer-process-is-enabled

Reply:
yup